squiggy/api/course_controller.py

```diff
@@ -0,0 +1,39 @@
+"""
+Copyright ©2021. The Regents of the University of California (Regents). All Rights Reserved.
+
+Permission to use, copy, modify, and distribute this software and its documentation
+for educational, research, and not-for-profit purposes, without fee and without a
+signed licensing agreement, is hereby granted, provided that the above copyright
+notice, this paragraph and the following two paragraphs appear in all copies,
+modifications, and distributions.
+
+Contact The Office of Technology Licensing, UC Berkeley, 2150 Shattuck Avenue,
+Suite 510, Berkeley, CA 94720-1620, (510) 643-7201, otl@berkeley.edu,
+http://ipira.berkeley.edu/industry-info for commercial licensing opportunities.
+
+IN NO EVENT SHALL REGENTS BE LIABLE TO ANY PARTY FOR DIRECT, INDIRECT, SPECIAL,
+INCIDENTAL, OR CONSEQUENTIAL DAMAGES, INCLUDING LOST PROFITS, ARISING OUT OF
+THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION, EVEN IF REGENTS HAS BEEN ADVISED
+OF THE POSSIBILITY OF SUCH DAMAGE.
+
+REGENTS SPECIFICALLY DISCLAIMS ANY WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. THE
+SOFTWARE AND ACCOMPANYING DOCUMENTATION, IF ANY, PROVIDED HEREUNDER IS PROVIDED
+"AS IS". REGENTS HAS NO OBLIGATION TO PROVIDE MAINTENANCE, SUPPORT, UPDATES,
+ENHANCEMENTS, OR MODIFICATIONS.
+"""
+
+from flask import current_app as app
+from flask_login import current_user
+from squiggy.lib.http import tolerant_jsonify
+from squiggy.lib.util import eb_environment
+from squiggy.models.canvas import Canvas
+
+
+@app.route('/api/canvas/all_domains')
+def all_canvas_domains():
+    # For purpose of user-friendly dev-auth form, we allow anonymous access in non-production.
+    eb_env = eb_environment() or ''
+    if 'prod' in eb_env and not current_user.is_admin:
```

Comment:
@johncrossman Would it make more sense to have this conditional dependent on the DEVELOPER_AUTH_ENABLED config rather than eb_environment? I'm thinking it might help with turning the feature on/off for testing, etc.

Reply:
Absolutely. PR amended.